Cython/Build/Inline.py

```diff
@@ -109,19 +109,175 @@ def _get_build_extension():
     return build_extension
 
 
+def cython_inline_raw(
+        code,
+        preprocessing_fn=None,
+        transform_code_fn=None,
+        module_cache_dir=os.path.join(get_cython_cache_dir(), 'inline'),
+        c_include_dirs=None,
+        distutils_extra_compile_args=None,
+        cythonize_include_dirs=None,
+        cythonize_compiler_directives=None,
+        cythonize_kwargs=None,
+        force=False,
+        module_name_base="_cython_inline_raw_",
+        cache_key=None):
+    """Compiles code to a python extension and returns the imported module.
+
+    A preprocessing function to be run on the code may be supplied in the form:
+
+        preprocessed_code, module_name, transform_code_args = \
+            preprocessing_fn(code)
+
+    A code transformation function to be run before compilation may also be
+    supplied in the form:
+
+        transformed_code, c_include_dirs, distutils_extra_compile_args = \
+            transform_code(code, c_include_dirs, distutils_extra_compile_args)
+    """
+
+    # set reasonable default values for undefined arguments
+    c_include_dirs = c_include_dirs if c_include_dirs is not None else []
+    distutils_extra_compile_args = distutils_extra_compile_args \
+        if distutils_extra_compile_args is not None else []
+    cythonize_include_dirs = cythonize_include_dirs \
+        if cythonize_include_dirs is not None else ['.']
+    cythonize_compiler_directives = cythonize_compiler_directives \
+        if cythonize_compiler_directives is not None else {}
+    cythonize_kwargs = cythonize_kwargs if cythonize_kwargs is not None else {}
+
+    # Fast path if this has been called in this session with the same parameters
+    preprocessing_code = preprocessing_fn.__code__ if preprocessing_fn else None
+    transform_code_code = transform_code_fn.__code__ if transform_code_fn else None
+    cache_key = (code, sys.version_info, sys.executable, Cython.__version__,
+        preprocessing_code, transform_code_code, tuple(c_include_dirs),
+        tuple(distutils_extra_compile_args), tuple(cythonize_include_dirs),
+        tuple(cythonize_compiler_directives), tuple(cythonize_kwargs.items()),
+        cache_key)
+    module = cython_inline_raw.cache.get(cache_key)
+    if module and not force:
+        return module
+
+    if preprocessing_fn:
+        code, module_name, transform_code_args = preprocessing_fn(code)
```

Comment:
Is there a need to return module_name? (In particular, retrieving the module from sys.modules when the cache was not hit above seems like a bug.)